core/primitives/resources/active_resources.hpp

```diff
@@ -30,7 +25,16 @@ namespace fc::primitives {
                                  const WorkerResources &resources,
                                  const ActiveResources &active);
 
+    inline void setMemoryUsedMin(uint64_t memory) {
+      std::shared_lock<std::shared_mutex> lock(mutex_);
```

Comment:
```suggestion
      std::unique_lock lock{mutex_};
```
or ignore, if test (the only usage) is safe